cpmpy/expressions/core.py

```diff
@@ -550,17 +550,10 @@ def get_bounds(self):
             lbs, ubs = zip(*[get_bounds(x) for x in self.args])
             return sum(lbs), sum(ubs)
         elif self.name == 'wsum':
+            weights, vars = self.args
+            bounds = np.array([x for x in zip(*[np.multiply(w, get_bounds(fvar)) for w, fvar in zip(weights, vars)])])
```

Comment:
Be sure to check that the bounds are ints (or make them ints), as in cases like in issue #243, when we have weights with float numbers, the bounds become float too and it will crash

Reply:
That is not really the task of the `get_bounds()` function I think, but rather when we construct the new variable.
@Wout4 You can write this even a little more efficient as follows:
```python
var_bounds = np.array(list(zip(*[get_bounds(arg) for arg in vars])))
bounds = var_bounds * weights
....
```

Reply:
oke so for now get_bounds can return floats, since that might be useful to get a more accurate calculation. I changed the creation of intvars to call math.floor and math.ceil